File: libs/dotnet/SquareSums/NodesSortingFacade.cs
using System;

namespace SquareSums
{
    public class NodesSortingFacade: INodesSorting
    {
        private readonly NativeNodesSorting _sorting;
        
        public NodesSortingFacade(Path? path, int maxN)
        {
            _sorting = new NativeNodesSorting(path, maxN);
        }
        
        public void SortNodes(Span<Node> nodes)
        {
            _sorting.SortNodes(nodes);
        }
    }
}

File: libs/go/pkg/nodes_sorting_facade.go
package sums

type NodesSortingFacade struct {
	sorting *NodesSortingNative
}

func NewNodesSortingFacade(path *Path, maxN int) *NodesSortingFacade {
	return &NodesSortingFacade{
		sorting: NewNodesSortingNative(path, maxN),
	}
}

func (rcvr *NodesSortingFacade) SortNodes(nodes []*Node) {
	rcvr.sorting.SortNodes(nodes)
}

type NodesSortingFactory = func(path *Path, maxN int) NodesSortingFacade
